Lesson32/Kadir/my-app/src/App.jsx

```diff
@@ -1,27 +1,49 @@
 import "./App.css";
 import { SearchBar } from "./components/SearchBar";
-import {useEffect, useState} from "react";
-import {RecipeList} from "./components/RecipeList"
-import {BD_SEARCH_BASE_URL } from "./urls";
+import { useEffect, useState } from "react";
+import { RecipeList } from "./components/RecipeList";
+import { BD_SEARCH_BASE_URL } from "./urls";
+import { RecipeDetail } from "./components/RecipeDetail";
 
 export const App = () => {
-  const [recipes, setRecipes]= useState([]);
+  const [recipes, setRecipes] = useState([]);
+  const [selectedRecipe, setSelectedRecipe] = useState();
+
+
+  const handleRecipeSelect = (recipe) => {
+    setSelectedRecipe(recipe);
+  };
+
+  const handleHomeClick = () => {
+    setSelectedRecipe(null);
+    window.location.reload();
+  };
+
 
   useEffect(() => {
-    fetch(BD_SEARCH_BASE_URL )
-    .then((response)=>response.json())
-    .then(data => {
-      if (data.meals) return setRecipes(data.meals);
-    })
-    .catch(error => setRecipes([]));
+    fetch(BD_SEARCH_BASE_URL)
+      .then((response) => response.json())
+      .then((data) => {
+        if (data.meals) return setRecipes(data.meals);
+      })
+      .catch((error) => setRecipes([]));
   }, []);
-  
+
+
   return (
     <div className="container">
-    <header>Recipe Search App</header>
-    <SearchBar setRecipes={setRecipes}/>
-    <RecipeList recipes={recipes} />
-    
+      <header onClick={handleHomeClick}>Recipe Search App</header>
+      <SearchBar setRecipes={setRecipes} />
+      <RecipeList
+        recipes={recipes}
+        handleRecipeSelect={handleRecipeSelect}
+        selectedRecipe={selectedRecipe}
+      />
+      {selectedRecipe ? (
+        <RecipeDetail selectedRecipe={selectedRecipe} />
+        ) : (
+        <p>Select a recipe to view details.</p>
```

Comment:
I'd actually not display this error message if no recipe is selected. 
```suggestion
        <></>
```